pkg/output/output.go

```diff
@@ -0,0 +1,65 @@
+package output
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+// Outputter is an interface for any object which can write the processed output.
+type Outputter interface {
+	// Write is used to write content of a file. The caller must ensure that the parent directory already exists.
+	Write(parent string, name string, contents string, permissions os.FileMode) error
```

Comment:
can we make this based on line-by-line processing (also on reading)? I'm really afraid of blowing up peoples memory with log files here.

Reply:
I've changed it to an line-by-line processor.